packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx

```diff
@@ -225,24 +230,48 @@ export const ContentPanel: FC = () => {
                                 activeEditorTab === EditorTabs.VISUALIZATION
                             }
                         >
-                            {queryResults?.results && barChartConfig && (
-                                <BarChart
-                                    data={queryResults}
-                                    config={barChartConfig}
-                                    isLoading={isLoading}
-                                    style={{
-                                        // NOTE: Ensures the chart is always full height
-                                        display:
-                                            selectedChartType ===
-                                            ChartKind.VERTICAL_BAR
-                                                ? 'block'
-                                                : 'none',
-                                        height: debouncedInputSectionHeight,
-                                        width: '100%',
-                                        flex: 1,
-                                    }}
-                                />
-                            )}
+                            {queryResults?.results &&
+                                barChartConfig &&
+                                selectedChartType ===
+                                    ChartKind.VERTICAL_BAR && (
+                                    <BarChart
+                                        data={queryResults}
+                                        config={barChartConfig}
+                                        isLoading={isLoading}
+                                        style={{
+                                            // NOTE: Ensures the chart is always full height
+                                            display:
+                                                selectedChartType ===
+                                                ChartKind.VERTICAL_BAR
+                                                    ? 'block'
+                                                    : 'none',
+                                            height: debouncedInputSectionHeight,
+                                            width: '100%',
+                                            flex: 1,
+                                        }}
+                                    />
+                                )}
+
+                            {queryResults?.results &&
+                                pieChartConfig &&
+                                selectedChartType === ChartKind.PIE && (
```

Comment:
if you remove this `selectedChartType === ChartKind.PIE` you avoid the chart re-rendering every time you switch the type.
This can also be applied on the bar chart above ^

Reply:
Ah, because you have that conditional visibility? 👌

Reply:
exactly 🙌